com/win32comext/axscript/client/error.py

```diff
@@ -114,7 +114,10 @@ def _BuildFromSyntaxError(self, site, exc, tb):
         except:
             msg = f"Unknown Error ({value})"
         try:
-            (filename, lineno, offset, line) = value[1]
+            filename = value[1][0]
+            lineno   = value[1][1]
+            offset   = value[1][2]
+            line     = value[1][3]
```

Comment:
Pretty sure this entire function can be simplified to:
```python
    def _BuildFromSyntaxError(self, exc: SyntaxError) -> None:
        # Some of these may be None, which upsets us!
        offset = exc.offset or 0
        line = exc.text or ""

        self.description = FormatForAX(str(exc))
        self.lineno = exc.lineno
        self.colno = offset - 1
        self.linetext = ExpandTabs(line.rstrip())
```

Reply:
Yes. This is nicely simplified and elegant. I have tested with Python 3.10.2 and 3.12.2 and I am getting the expected behavior from my C++ code when a Python script executed through Windows Scripting Host has a syntax error.